Add -Wno-deprecated-declarations to demo when NDK >=26 is used

cmannett85-arm opened the pull request:
## Description
Compiles the demo APK with -Wno-deprecated-declarations when the NDK is v26 or over, otherwise it fails to compile the fmt core header.  In the LLVM version used in the NDK, using `std::char_traits` with non-std types is marked as deprecated.

For NDKs less than v26, there is no change in behaviour.

baldurk commented on util/test/demos/CMakeLists.txt:
```diff
@@ -73,8 +87,14 @@ if(BUILD_ANDROID)
+
+    # The version of fmt used will not build due to std::char_traits support of non-std types being
+    # deprecated in the version of LLVM used by NDK v26
+    if("${ANDROID_NDK_VERSION}" VERSION_GREATER_EQUAL "26.0.0")
```
I got bitten by these version compares recently, cmake's `VERSION_GREATER_EQUAL` was added in a later version than what we support. You need to formulate this as `NOT X VERSION_LESS Y` instead :(.

cmannett85-arm replied:
Ah yes, you're right it was brought in for v3.7.

baldurk reviewed (changes requested):
Looks good! I've noted one compatibility issue only.

cmannett85-arm committed:
Add -Wno-deprecated-declarations to demo when NDK >=26 is used

Compiles the demo APK with -Wno-deprecated-declarations when the NDK is v26 or over, otherwise it fails to compile the fmt core header.  In the LLVM version used in the NDK, using std::char_traits with non-std types is marked as deprecated.
